Improved pan extension

Fueredoriku opened the pull request:
- Pan now blocks explosion damage if aimed correctly
![panSplosion](https://github.com/user-attachments/assets/54718f10-a223-4c01-bf81-00b4a39246d3)
- Pan skating speed increased a lot
- Add pan-shot! popup text when blocking damage

![panShot](https://github.com/user-attachments/assets/6715c150-7650-441f-8fa9-c5f01115bfa6)

toberge reviewed (changes requested):
Nice that it finally blocks explosions 🎉 (though sometimes not quite)

However, currently it seems that bullets are blocked by the huge explosion catchers as well, which makes it impossible to hit someone from the front 🙃 
![image](https://github.com/user-attachments/assets/60a378ad-cc3c-4d28-ad6b-fb8e629b6cdf)

toberge commented on Assets/Scripts/Augment/AugmentImplementations/Pan.cs:
```diff
@@ -22,30 +26,48 @@ void Start()
     {
         if (playerMovement)
-            if (playerMovement.StateIsAir)
+            if (playerMovement.StateIsAir && playerMovement.IsCrouching)
```
at this point the `if` pyramid looks a little odd

Fueredoriku replied:
Hehehe, I concede that the pyramid should go, I will miss it tho 🫡

toberge commented on Assets/Scripts/Augment/AugmentImplementations/PanHitbox.cs:
```diff
@@ -0,0 +1,20 @@
+public class PanHitbox : HitboxController
+{
+    [SerializeField]
+    private HealthController panHealth;
```
Superfluous. The superclass already has a public health member.

Fueredoriku replied:
They have two different purposes, the panhealth is for the panshotemitter, not for tracking actual damage!

toberge replied:
Oh okay

Fueredoriku committed:
Add improved pan extension
- Pan now blocks explosion damage if aimed correctly
- Pan skating speed increased a lot
- Add pan turbo slide by leaping into skate
- Add pan-shot! popup text when blocking damage

Fueredoriku committed:
Fix issue where decals would attach to and follow hitboxes

toberge reviewed (approved):
Works great now!
